driver-mapping/src/main/java/com/datastax/driver/mapping/PropertyMapper.java

```diff
@@ -0,0 +1,196 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.mapping;
+
+import com.datastax.driver.core.Metadata;
+import com.datastax.driver.core.TypeCodec;
+import com.datastax.driver.mapping.annotations.*;
+import com.google.common.reflect.TypeToken;
+
+import java.beans.PropertyDescriptor;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+import java.util.Collection;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkState;
+
+/**
+ * Maps a Java bean property to a table column or a UDT field.
+ * <p>
+ * Properties can be either accessed through getter and setter pairs,
+ * or by direct field access, depending on what is available in the
+ * entity/UDT class.
+ */
+class PropertyMapper {
```

Comment:
Not sure this matters, but I noticed that `ColumnMapper` was generic, where its replacement, `PropertyMapper,` isn't.  Is there any particular reason why it was removed?  I see that it really wasn't needed in the first place, since the callers don't particularly deal with the type that is being mapped directly.